kani-compiler/src/codegen_cprover_gotoc/overrides/hooks.rs

```diff
@@ -315,6 +316,54 @@ impl GotocHook for IsAllocated {
     }
 }
 
+/// This is the hook for the `kani::float::float_to_int_in_range` intrinsic
+/// TODO: This should be replaced by a Rust function instead so that it's
+/// independent of the backend
+struct FloatToIntInRange;
+impl GotocHook for FloatToIntInRange {
+    fn hook_applies(&self, _tcx: TyCtxt, _instance: Instance) -> bool {
+        unreachable!("{UNEXPECTED_CALL}")
+    }
+
+    fn handle(
+        &self,
+        gcx: &mut GotocCtx,
+        instance: Instance,
+        mut fargs: Vec<Expr>,
+        assign_to: &Place,
+        target: Option<BasicBlockIdx>,
+        span: Span,
+    ) -> Stmt {
+        assert_eq!(fargs.len(), 1);
+        let float = fargs.remove(0);
+        let target = target.unwrap();
+        let loc = gcx.codegen_span_stable(span);
+
+        let generic_args = instance.args().0;
+        let RigidTy::Float(float_ty) = generic_args[0].expect_ty().kind().rigid().unwrap().clone()
```

Comment:
```suggestion
        let RigidTy::Float(float_ty) = generic_args[0].expect_ty().kind().rigid().unwrap()
```

Reply:
This is what I had initially, but this results in an error:
```
error[E0716]: temporary value dropped while borrowed
   --> kani-compiler/src/codegen_cprover_gotoc/overrides/hooks.rs:343:40
    |
343 |         let RigidTy::Float(float_ty) = generic_args[0].expect_ty().kind().rigid().unwrap()
    |                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ creates a temporary value which is freed while still in use
...
346 |         };
    |          - temporary value is freed at the end of this statement
...
351 |             *float_ty,
    |             --------- borrow later used here
```
It works if I add a deref:
```suggestion
        let RigidTy::Float(float_ty) = *generic_args[0].expect_ty().kind().rigid().unwrap()
```
but I think this is equivalent to the `clone()` version.

Reply:
Yes, you are correct that you need to dereference it.

Reply:
Done.